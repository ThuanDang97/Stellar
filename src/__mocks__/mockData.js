export const TOOLTIP_LOREM =
  'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'

export const SELECT_OPTIONS = [
  { value: '1', name: 'Pay monthly' },
  { value: '3', name: 'Pay 3 monthly' },
  { value: '6', name: 'Pay 6 monthly' },
  { value: '12', name: 'Pay annually' },
]

export const USER_FULL_NAME = 'John Wick'

export const AVATAR_DEFAULT = {
  firstName: 'John',
  lastName: 'Wick',
}

export const AVATAR_IMAGE = {
  imageUrl: 'https://picsum.photos/20',
}

export const NETFLIX_REPORT = {
  name: 'netflix',
  period: 'report monthly',
}

export const GOAL_ITEM = {
  title: 'Improve My Credit Score',
  iconUrl: '/icons/piggy-bank',
}

export const BANK_NUMBER = '123456789111'

export const BANK_NAME = 'Chase'

export const INPUT_VALUES = {
  email: 'test@gmail.com',
  text: 'example',
  password: 'example1password',
  image:
    'data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e',
}

export const MEDIA_IMAGES = {
  netflix: '/images/netflix.png',
  hulu: '/images/hulu.png',
}

export const ADD_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Holo',
    amount: 77,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const MY_GOALS = [
  {
    id: '1',
    goal: 'Buy a car',
    scoreNeeded: 703,
    pointNeeded: 'READY',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    goal: 'Buy a house',
    scoreNeeded: 720,
    pointNeeded: '23 PTS AWAY',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 3434543645,
  },
]

export const LINKED_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    draftDate: '1st',
    status: 'Unpaid',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Hulu',
    amount: 32,
    draftDate: '4th',
    status: 'Paid',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const EXITS_EMAIL = 'thanh@endurance.com'

export const NUMBER_BANK = [
  {
    id: '1',
    bankNumber: '12435324234',
    bankName: 'Chase',
  },
]

export const ACCOUNT_LOGIN = [
  {
    userId: '1',
    email: 'khanhtam304n@gmail.com',
    password: 'Khanhtam@123',
  },
  {
    userId: '2',
    email: 'khanhtam3041n@gmail.com',
    password: 'Khanhtam@123',
  },
]

export const USER = {
  userId: '4',
  userName: 'Khanh Tam',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

export const GOAL = ['g1', 'g2']

export const MARKETING = ['m1', 'm2']

export const USER_ID = '123'

export const DATE_TIMER = new Date('08/28/2022')

export const USER_GOAL = { userId: USER_ID, goals: GOAL }

export const USER_MARKETING = { userId: USER_ID, goals: MARKETING }

export const ADDRESS = {
  address: '123 street',
  phone: '1231231231',
}

export const SSN_NUMBER = '123123123'

export const SUBSCRIPTION_PLAN = 'flex'

export const USER_SUBSCRIPTION_PLAN = {
  userId: USER_ID,
  subscriptionPlan: SUBSCRIPTION_PLAN,
}

export const CHART_SCORE = {
  data: [
    { month: 'M', score: 720 },
    { month: 'A', score: 740 },
    { month: 'M', score: 768 },
    { month: 'J', score: 785 },
    { month: 'J', score: 810 },
    { month: 'A', score: 830 },
  ],
}

export const LINK_TOKEN = 'sand-box-123'

export const BANK_ID = 'id_45'

export const BANK_INFO = {
  id: BANK_ID,
  bankNumber: BANK_NUMBER,
  bankName: BANK_NAME,
}
export const FICO_DATE_UPDATE = 'Last Updated Aug, 12, 2021'

export const NOTIFICATIONS = [
  {
    id: '1',
    title: 'Credit Score Update',
    desc: 'Your credit score increased by...',
    dateTime: '1 hours',
    isCheck: false,
  },
  {
    id: '2',
    title: 'Bill Pay',
    desc: 'We noticed you missed a payment...',
    dateTime: '3 days ago',
    isCheck: false,
  },
  {
    id: '3',
    title: 'Offer',
    desc: 'Look like you may quality for...',
    dateTime: '10 days ago',
    isCheck: false,
  },
  {
    id: '4',
    title: 'Offer',
    desc: 'Look like you may quality for offer',
    dateTime: '11 days ago',
    isCheck: false,
  },
]

export const APPLICATION_INFO = {
  title: 'Netflix',
  image: '/images/netflix.png',
}

export const APPLICATION_CONNECT_ACCOUNT = {
  userId: '1',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

// Min score: 300 - Max score 850
export const FICO_SCORE = 500

export const IDENTITY = {
  firstName: 'John',
  lastName: 'Wick',
  dateOfBirth: '02/02/2000',
  ssn: 123123123,
}

export const SSN_USER = {
  userId: USER_ID,
  ssn: SSN_NUMBER,
}

export const USER_IDENTITY = {
  userId: USER_ID,
  identity: IDENTITY,
}

export const ADDRESS_USER = {
  address: '604 Nui Thanh',
  phoneNumber: '+10949328492',
}

export const FICO_UNLOCK_SCORE = 708

export const LIST_BANKS_NUMBER = [
  { id: 'Chase', bankNumber: '423432432423232' },
  { id: 'Bank of America', bankNumber: '111122223333000' },
  { id: 'CitiBank Online', bankNumber: '111122223333111' },
]

export const INFO_PAYMENTS = {
  id: '123',
  money: '15.99',
  bankName: 'Chase',
  payDate: new Date(),
  numberPaymentsMonth: '3',
  appInfo: {
    title: 'Netflix',
    image: '/images/netflix.png',
  },
  status: 'UNPAID',
}

export const BILL_INFO = {
  title: 'Netflix',
  amount: 10,
}

export const ADDED_BILL_ID = '1234'

export const USER_DETAIL = {
  id: '123',
  amount: 10,
  draftDate: 31,
  bankNumber: BANK_NUMBER,
}

export const EDIT_LINK_BILL = {
  id: '123',
  amount: 10,
}
