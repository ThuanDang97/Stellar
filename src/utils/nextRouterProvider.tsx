import { NextRouter } from 'next/router'
import { RouterContext } from 'next/dist/shared/lib/router-context'

const createMockRouter = (router?: Partial<NextRouter>): NextRouter => ({
  basePath: '',
  pathname: '/',
  route: '/',
  query: {},
  asPath: '/',
  back: jest.fn(),
  beforePopState: jest.fn(),
  prefetch: jest.fn(),
  push: jest.fn(),
  reload: jest.fn(),
  replace: jest.fn(),
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  isFallback: false,
  isLocaleDomain: false,
  isReady: true,
  defaultLocale: 'en',
  domainLocales: [],
  isPreview: false,
  ...router,
})

export const NextRouterProvider = ({
  children,
  router = { query: {} },
}: {
  children: React.ReactNode
  router: Partial<NextRouter>
}) => (
  <RouterContext.Provider value={createMockRouter(router)}>
    {children}
  </RouterContext.Provider>
)
