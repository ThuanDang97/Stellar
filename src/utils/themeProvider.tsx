import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
