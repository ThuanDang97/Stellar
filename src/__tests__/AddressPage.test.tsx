import React from 'react'
import { cleanup, fireEvent, screen, waitFor } from '@testing-library/react'
import Router from 'next/router'

// Utils
import { renderWithTheme } from '@utils/themeProvider'
import { NextRouterProvider } from '@utils/nextRouterProvider'

// Pages
import AddressPage from '@pages/confirm-address'

//  Mocks
import { USER_ID } from '@mocks/mockData'

const query = { userid: USER_ID }
describe('AddressPage', () => {
  beforeEach(() => {
    cleanup()
    renderWithTheme(
      <NextRouterProvider router={{ query }}>
        <AddressPage />
      </NextRouterProvider>,
    )
  })

  it('should render addressForm', () => {
    const addressForm = screen.getByTestId('addressForm')

    expect(addressForm).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = renderWithTheme(<AddressPage />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should onChange value when input value', async () => {
    const input = screen.getAllByTestId('iconInput') as HTMLInputElement[]
    fireEvent.change(input[0], { target: { value: 'Cheese pocket' } })
    expect(input[0].value).toBe('Cheese pocket')
  })

  test('should change value phone input', async () => {
    const input = screen.getAllByTestId('iconInput') as HTMLInputElement[]
    fireEvent.change(input[1], { target: { value: '1231231231' } })
    expect(input[1].value).toBe('(123) 123-1231')
  })

  it('should render errorMessage', () => {
    const input = screen.getAllByTestId('iconInput') as HTMLInputElement[]
    fireEvent.change(input[0], { target: { value: 'Cheese pocket' } })
    fireEvent.change(input[1], { target: { value: '123123121' } })

    const button = screen.getByTestId('button')

    fireEvent.click(button)

    waitFor(() => {
      const error = screen.getByTestId('errorMessage')
      expect(error).not.toBeInTheDocument()
    })
  })

  it('should checkout new page when submit success', () => {
    const input = screen.getAllByTestId('iconInput') as HTMLInputElement[]
    fireEvent.change(input[0], { target: { value: 'Cheese pocket' } })
    fireEvent.change(input[1], { target: { value: '123123121' } })

    const button = screen.getByTestId('button')

    fireEvent.click(button)

    waitFor(() => {
      expect(Router.push).toHaveBeenCalledWith('/dashboard')
    })
  })
})
