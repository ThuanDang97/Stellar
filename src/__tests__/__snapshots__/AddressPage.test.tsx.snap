// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__WrapperContent-sc-rqe5sa-6 bTYFgu gVWyht"
  >
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 TitleStyled__TitlePageStyled-sc-t58pjn-1 fkzkBU ONSni ilgUUT"
      color="#242A30"
      font-family="AdobeCleanExtraBold"
      font-size="36"
    >
      Where do you live?
    </h1>
    <p
      class="TextStyled-sc-9xia41-0 wWWum"
    >
      <span
        class="TextStyled-sc-9xia41-0 dphVoX"
        color="#242A30"
        font-size="16"
      >
        Your address is used to verify your identity.
      </span>
    </p>
    <form
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 uWEvw eFQizc"
      data-testid="addressForm"
      display="flex"
      novalidate=""
      width="100%"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperInputStyled-sc-9gj86m-1 lkGwqj brcjID"
        color="#203F59"
        data-testid="inputIcon"
        display="block"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 fTsvHV kHphsF"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="icon"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <input
          class="InputStyled-sc-1k09cqb-0 InputStyled__InputItemStyled-sc-9gj86m-0 hnxmhM NDCRX"
          data-testid="iconInput"
          name="address"
          placeholder="Home Address"
          type="text"
          value=""
        />
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperInputStyled-sc-9gj86m-1 iZFyqP brcjID"
        color="#203F59"
        data-testid="inputIcon"
        display="block"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 fTsvHV kHphsF"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="icon"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <input
          class="InputStyled-sc-1k09cqb-0 InputStyled__InputItemStyled-sc-9gj86m-0 hnxmhM NDCRX"
          data-testid="iconInput"
          name="phone"
          placeholder="Phone Number"
          type="tel"
          value=""
        />
      </div>
      <button
        class="ButtonStyled-sc-1ys9v52-0 iAfDLq"
        data-testid="button"
        disabled=""
        type="submit"
      >
        Continue
      </button>
    </form>
    <p
      class="TextStyled-sc-9xia41-0 gtHdGO"
    >
      <span
        class="TextStyled-sc-9xia41-0 inBuZW"
        color="#F2406E"
        font-size="13"
      >
        This will not affect your credit score. We donâ€™t store your sensitive information.
      </span>
      <a
        class="LinkStyled__ButtonLinkStyled-sc-626068-1 bpcOwZ"
        data-testid="link"
        display="inline"
        font-size="14"
        font-weight="800"
        href="/sign-up"
        text-decoration="underline"
      >
        Learn more
      </a>
      <span
        class="TextStyled-sc-9xia41-0 inBuZW"
        color="#F2406E"
        font-size="13"
      >
        about our security.
      </span>
    </p>
  </div>
</DocumentFragment>
`;
